### PR TITLE
adds initial layout and styling

### PR DESCRIPTION
**Pull request by astewes**
-  Adds initial layout for tracker.tags state

tags

No tags to display:
wells

### CODE REVIEW

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+ */
+
+class TrackerTagsController{
+  constructor($state, $scope, myTrackerApi, $uibModal) {
+    this.$state = $state;
+    this.$scope = $scope;
+    this.myTrackerApi = myTrackerApi;
+    this.$uibModal = $uibModal;
+
+    this.preferredTags = [
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js]: I thought we were going to use the mocks APIs?

File: cdap-ui/app/features/tracker/templates/tags.html [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      <table class="table table-striped">
+        <thead>
+          <tr>
+            <th>Preferred Tags</th>
+            <th>Tagged Entities</th>
+            <th>Action</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr ng-repeat="tag in TagsController.preferredTags">
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/templates/tags.html]: Can we have all possible scope bounded variables to be single binded? Not sure what could change in this list. Are we polling for anything?

**Comment by ajainarayanan** [comment]
@astewes Also what happens when we have no tags? Do we have a placeholder to say that there are no tags added to the system? Will we even have that as a usecase?

**Comment by astewes** [comment]
@ajainarayanan according to the mocks, when there are no tags, an empty table is rendered

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+class TrackerTagsController{
+  constructor($state, $scope, myTrackerApi, $uibModal) {
+    this.$state = $state;
+    this.$scope = $scope;
+    this.myTrackerApi = myTrackerApi;
+    this.$uibModal = $uibModal;
+
+    this.tagList = {
+      preferred: 0,
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js]: Any reason why we need a separate count for preferred tags? Could it be `this.tagList.preferredTags.length`?

Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js]: I think because of this I am only seeing empty preferred/user tags section

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        {
+          'key': 'tag5',
+          'total_entity': 28
+        }
+      ]
+    };
+
+    this.goToTag = function(event, tag) {
+      event.stopPropagation();
+      $state.go('search.objectswithtags', {tag: tag});
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js]: This could probably be in the html template. This will prevent us from doing CMD + Click to open the search in a new tab in the browser.

File: cdap-ui/app/features/tracker/controllers/tags-ctrl.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        },
+        {
+          'key': 'tag5',
+          'total_entity': 0
+        }
+      ],
+      user: 0,
+      userTags: [
+        {
+          'key': 'tag1',
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/controllers/tags-ctrl.js:51]: This should be either `name` or navigation to search should use `key`. Right now we get an empty double quotes in the search page in place of name.

screen shot 2016-07-08 at 8 52 04 am

File: cdap-ui/app/features/tracker/templates/partial/add-preferred-tags-popover.html [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  <h3 class="modal-title">
+    <span>Add Preferred Tags</span>
+    <span class="fa fa-times pull-right" ng-click="$close()"></span>
+  </h3>
+</div>
+
+<div class="modal-body">
+  <div class="flex-box">
+    <div class="flex-item">
+      <div>Enter preferred tags here<br />
```
Comment (ajainarayanan) [inline cdap-ui/app/features/tracker/templates/partial/add-preferred-tags-popover.html]: This should be a textarea. The user should be able to type in the tags either comma separated or new line separated. We could still handle this in UI. May be in another PR but this needs to be a textarea according to mocks.

**Comment by nitinmotgi** [comment]
@ajainarayanan merge ?

**Comment by elfenheart** [comment]
👍  LGTM
